### PR TITLE
Questions reordering

### PR DESCRIPTION
**Pull request by pascalgrimaud**
---

Please make sure the below checklist is followed for Pull Requests.

-   [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
-   [ ] Tests are added where necessary
-   [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) was updated if necessary
-   [ ] Documentation is added/updated where necessary
-   [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (bellow reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: generators/server/prompts.js [hunk 1]
```diff
@@ -349,7 +348,7 @@ function askForOptionalItems() {
         }
     }
     choices.push({
-        name: 'Asynchronous messages using Apache Kafka',
+        name: 'Apache Kafka as asynchronous messages',
```
Comment (avdev4j) [inline generators/server/prompts.js:351]:
```suggestion
        name: 'Apache Kafka as asynchronous messages broker',
```

File: generators/app/prompts.js [hunk 2]
```diff
@@ -93,14 +93,14 @@ async function askForTestOpts() {
...
     const choices = [];
-    if (!this.skipServer) {
-        // all server side test frameworks should be added here
-        choices.push({ name: 'Gatling', value: 'gatling' }, { name: 'Cucumber', value: 'cucumber' });
-    }
     if (!this.skipClient) {
         // all client side test frameworks should be added here
-        choices.push({ name: 'Protractor **Deprecated**', value: 'protractor' });
         choices.push({ name: 'Cypress', value: 'cypress' });
+        choices.push({ name: '[DEPRECATED] Protractor', value: 'protractor' });
```
Comment (avdev4j) [inline generators/app/prompts.js:99]: Why changing the order this way?

File: generators/app/prompts.js [hunk 3]
```diff
@@ -51,12 +51,12 @@ async function askForApplicationType() {
             name: 'Monolithic application (recommended for simple projects)',
         },
         {
-            value: 'microservice',
-            name: 'Microservice application',
+            value: 'gateway',
```
Comment (avdev4j) [inline generators/app/prompts.js:54]: Why putting gateway before microservice?

Comment (pascalgrimaud) [reply generators/app/prompts.js:54]: easier to understand IMO
- to not confuse between Microservice application and Microservice gateway
- the 1st and 2nd option have front
- the 3rd and 4th backend only

Comment (avdev4j) [reply generators/app/prompts.js:54]: ok, make sens now.

Comment (pascalgrimaud) [reply generators/app/prompts.js:54]: it's because I select the wrong option several times... :-)

File: generators/app/prompts.js (hunk 2 above)
Comment (pascalgrimaud) [reply generators/app/prompts.js:99]: Front frameworks first, then Backend frameworks after
By experience, it's more valuable than Gatling and Cucumber

**Review by avdev4j — APPROVED** [review]

**Pull request merged**